src/compiler/typescript_to_rust_redesign/azle_generate/src/generators/canister_methods/method_body.rs

```diff
@@ -43,3 +47,54 @@ pub fn generate_call_to_js_function(fn_decl: &FnDecl) -> proc_macro2::TokenStrea
         ).unwrap();
     }
 }
+
+/// Generates the return expression for a canister method body
+///
+/// # Context
+///
+/// * `_azle_final_return_value: boa_engine::JsValue` - The value to be returned
+///    unless this is a ManualReply method.
+/// * `_azle_boa_context: &mut boa_engine::Context` - The current boa context
+fn generate_return_expression(fn_decl: &FnDecl) -> proc_macro2::TokenStream {
+    if fn_decls::is_manual(fn_decl) {
+        return quote! {
+            ic_cdk::api::call::ManualReply::empty()
+        };
+    }
+
+    let return_type = fn_decls::get_canister_method_return_type(fn_decl);
+
+    if type_is_emptyish(return_type) {
+        return quote! {
+            return;
+        };
+    }
+
+    quote! {
+        _azle_final_return_value.azle_try_from_js_value(&mut _azle_boa_context).unwrap()
+    }
+}
+
+/// Returns true if the return type is `empty`, `never`, `null`, `void`, or `undefined`.
+/// Otherwise returns false.
+fn type_is_emptyish(ts_type_option: Option<&TsType>) -> bool {
```

Comment:
Get rid of never and undefined, make a comment TODO to consider that functionality

Reply:
:+1: Done!